### PR TITLE
Feat adding healthCheck

### PR DESCRIPTION
**Pull request by roggervalf**
Exploring a feature to verify if a queue is still connected

### CODE REVIEW

File: src/classes/redis-connection.ts [hunk 1]
```diff
@@ -138,6 +138,16 @@ export class RedisConnection extends EventEmitter {
     }
   }
 
+  async healthCheck(): Promise<boolean> {
+    try {
+      await this._client.info();
+      return true;
+    } catch (error) {
+      this.emit('error', error);
```
Comment (manast) [inline src/classes/redis-connection.ts:146]: emit here is a side-effect, we should not create side-effects in a function that is supposed to just return true/false

**Review by manast — COMMENTED** [review]
Thanks for the PR. I think that it is better to keep a utility function like this outside of the library since it is easy to implement and should not really be part of the scope of BullMQ. It we were to use that same function internally it would a different thing of course.

**Comment by roggervalf** [comment]
> Thanks for the PR. I think that it is better to keep a utility function like this outside of the library since it is easy to implement and should not really be part of the scope of BullMQ. It we were to use that same function internally it would a different thing of course.

Let me see, I guess I can try something

**Comment by roggervalf** [comment]
I tried to use it to fix this test https://github.com/taskforcesh/bullmq/blob/master/src/test/test_connection.ts#L152 but sadly it doesn't work, I'll close this pr for now

**Pull request closed without merging**